### PR TITLE
Feature/25 integrate notebooks into docu

### PR DESCRIPTION
**Pull request by MarcoHuebner**
- update conf, index and deployment workflow to add notebooks to docu
- update dependencies
- adapt pre-commit hook: Remove step local documentation building

Addresses #25 

## Release Notes

- **New Features**
  - Enhanced documentation deployment process with the addition of Pandoc installation and improved notebook integration.
  - Added support for Jupyter Notebooks in documentation through the `nbsphinx` extension.

- **Documentation**
  - Updated README to clarify prerequisites for building documentation locally.
  - Introduced a new table of contents for notebooks in the documentation index for better navigation.

- **Bug Fixes**
  - Improved formatting and readability in the Geo-Visualization Example Notebook.

### CODE REVIEW

**Comment by pmayd** [comment]
@coderabbit review

File: .github/workflows/deploy-docs.yaml [hunk 1]
```diff
@@ -27,13 +27,24 @@
...
           poetry install --with dev
+      - name: Remove existing nb directory
+        run: |
+          if [ -d "docs/source/nb" ]; then
+            rm -rf docs/source/nb
+          fi
+      - name: Copy Notebook to docs
+        run: |
+          mkdir -p docs/source/nb
+          cp -r nb/* docs/source/nb/
```
[the remark this answers is not in the document]
Comment (MarcoHuebner) [reply .github/workflows/deploy-docs.yaml]: Included in https://github.com/CorrelAid/pystatis/pull/149/commits/4a1358c4e68a71c0d715b9e80347617cc866f9c3

File: .github/workflows/deploy-docs.yaml [hunk 2]
```diff
@@ -27,13 +27,24 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Install Pandoc
+        run: sudo apt-get install -y pandoc
```
[the remark this answers is not in the document]
Comment (MarcoHuebner) [reply .github/workflows/deploy-docs.yaml]: Included in https://github.com/CorrelAid/pystatis/pull/149/commits/4a1358c4e68a71c0d715b9e80347617cc866f9c3

File: pyproject.toml [hunk 3]
```diff
@@ -63,6 +63,7 @@ sphinx = "^7.0"
 sphinx-rtd-theme = "^2.0.0"
 urllib3 = {extras = ["appengine"], version = "^2.2.1"}
 pytest-vcr = "^1.0.2"
+nbsphinx = "^0.9.5"
```
[the remark this answers is not in the document]
Comment (MarcoHuebner) [reply pyproject.toml:66]: It is indeed listed in the conf.py, also at this point in time

**Comment by MarcoHuebner** [comment]
Built documentation can be found at https://correlaid.github.io/pystatis/dev/index.html

**Pull request merged**